rust/tools/authorization-logic/src/parsing/astconstructionvisitor.rs

```diff
@@ -49,14 +49,15 @@ fn construct_predicate(ctx: &PredicateContext) -> AstPredicate {
         None => Sign::Positive 
     };
     // Note that ID_all() in the generated antlr-rust code is buggy
-    // (because all {LEX_RULE}_all() generations are buggy),
+    // (because all {LEX_RULE}_all() generations are buggy
+    // by contrast to {PARSE_RULE}_all() generations which are fine),
     // so rather than using a more idomatic iterator, "while Some(...)" is
     // used here.
-    let name_ = ctx.ID(0).unwrap().get_text();
+    let name_ = ctx.ID().unwrap().get_text();
     let mut args_ = Vec::new();
-    let mut idx = 1;
-    while let Some(id) = ctx.ID(idx) {
-        args_.push(id.get_text());
+    let mut idx = 0;
+    while let Some(arg) = ctx.pred_arg(idx) {
+        args_.push(arg.get_text());
         idx += 1;
```

Comment:
Rather than having `mut` locals and using a `while` loop, would it be possible to refactor this to use `map` on `pred_arg` to produce a `Vec` and remove the mutation?

Reply:
fixed